utils_cv/detection/bbox.py

```diff
@@ -25,14 +30,66 @@ def __init__(self, left: int, top: int, right: int, bottom: int):
         self.standardize()
 
     @classmethod
-    def from_array(cls, arr: List[int]) -> "Bbox":
+    def from_array(cls, arr: List[int]) -> "_Bbox":
         """ Create a Bbox object from an array [left, top, right, bottom] """
         return _Bbox(arr[0], arr[1], arr[2], arr[3])
 
     @classmethod
-    def from_array_xywh(cls, arr: List[int]) -> "Bbox":
-        """ create a Bbox object from an array [left, top, width, height] """
-        return _Bbox(arr[0], arr[1], arr[0] + arr[2], arr[1] + arr[3])
+    def from_array_xywh(cls, arr: List[int]) -> "_Bbox":
+        """ Create a Bbox object from an array [left, top, width, height] """
+        return _Bbox(arr[0], arr[1], arr[0] + arr[2] - 1, arr[1] + arr[3] - 1)
+
+    @classmethod
+    def get_rect_from_binary_mask(cls, binary_mask: np.ndarray) -> List[int]:
+        """ Get the bounding box rectangle from a binary numpy mask.
+
+        Args:
+            binary_mask: boolean numpy array.  True indicates the pixel belongs
+                to the object in the mask, otherwise, False.
+        """
+        pos = np.where(binary_mask)
+        left = np.min(pos[1])
+        right = np.max(pos[1])
+        top = np.min(pos[0])
+        bottom = np.max(pos[0])
+        return [left, top, right, bottom]
+
+    @classmethod
+    def from_binary_mask(cls, binary_mask: np.ndarray) -> "_Bbox":
+        """ Create a Bbox object from a binary numpy mask """
+        return cls.from_array(cls.get_rect_from_binary_mask(binary_mask))
+
+    @classmethod
+    def from_mask(cls, mask: Union[np.ndarray, str, Path]) -> "List[_Bbox]":
+        """ Create a list of Bbox objects from a numpy mask
+
+        Assume the mask is grayscale with different values representing
+        different objects, 0 as background.
+        """
+        return [cls.from_binary_mask(bmask) for bmask in binarise_mask(mask)]
+
+    @classmethod
+    def hflip_rects(
+        cls,
+        rects: Union[torch.Tensor, np.ndarray, List[List[int]]],
+        width: int,
+    ) -> Union[torch.Tensor, np.ndarray, List[List[int]]]:
+        """ Flip rectangles horizontally.
+
+        Args:
+            rects: list of rectangles in the form of [left, top, right, bottom]
+            width: width of the image which the rectangles are from
+        """
+        islist = False
+        if isinstance(rects, list):
```

Comment:
Does this function need to support list and np.array, ie is this functionality used somewhere? If not, then remove code that's not needed to simplify this function.

Reply:
Removed this function